Sources/NIO/SocketChannel.swift

```diff
@@ -284,6 +284,10 @@ class BaseSocketChannel<T: BaseSocket>: SelectableChannel, ChannelCore {
     fileprivate func setOption0<T: ChannelOption>(option: T, value: T.OptionType) throws {
         assert(eventLoop.inEventLoop)
 
+        guard isOpen else {
+            throw ChannelError.alreadyClosed
```

Comment:
`.alreadyClosed` is defined as only `close` on already closed channel. This should be `ioOnClosedChannel`

Reply:
![screen shot 2018-03-20 at 3 42 04 pm](https://user-images.githubusercontent.com/624238/37665489-4ad277aa-2c55-11e8-91b0-0cf83703cd84.png)

Reply:
@weissi fixed